### ISSUE TITLE
fix crafted tool durability

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#156]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/156) by Meelis Mägi (Bitbucket: [Meelis Mägi](https://bitbucket.org/Meelis Mägi), ).**

----------------------------------------

Tools crafted with durability rite should take CraftParameters.Durability (=0.2) into account.

```
#!c++

diff -r 17c819ec88cd code/ryzom/server/src/entities_game_service/game_item_manager/game_item.cpp
--- a/code/ryzom/server/src/entities_game_service/game_item_manager/game_item.cpp       Fri May 23 19:13:36 2014 +0200
+++ b/code/ryzom/server/src/entities_game_service/game_item_manager/game_item.cpp       Tue May 27 11:28:37 2014 +0300
@@ -2895,13 +2895,13 @@
 
                        // tools
                        // SHEARS = pick for forage
-                       case ITEM_TYPE::SHEARS:                 return (uint32)CWeaponCraftParameters::ForageToolDurability;
-                       case ITEM_TYPE::AmmoTool:               return (uint32)CWeaponCraftParameters::AmmoCraftingToolDurability;
-                       case ITEM_TYPE::ArmorTool:              return (uint32)CWeaponCraftParameters::ArmorCraftingToolDurability;
-                       case ITEM_TYPE::JewelryTool:    return (uint32)CWeaponCraftParameters::JewelryCraftingToolDurability;
-                       case ITEM_TYPE::MeleeWeaponTool:return (uint32)CWeaponCraftParameters::MeleeWeaponCraftingToolDurability;
-                       case ITEM_TYPE::RangeWeaponTool:return (uint32)CWeaponCraftParameters::RangeWeaponCraftingToolDurability;
-                       case ITEM_TYPE::ToolMaker:              return (uint32)CWeaponCraftParameters::ToolCraftingToolDurability;
+                       case ITEM_TYPE::SHEARS:                 d = CWeaponCraftParameters::ForageToolDurability;                               break;
+                       case ITEM_TYPE::AmmoTool:               d = CWeaponCraftParameters::AmmoCraftingToolDurability;                 break;
+                       case ITEM_TYPE::ArmorTool:              d = CWeaponCraftParameters::ArmorCraftingToolDurability;                break;
+                       case ITEM_TYPE::JewelryTool:    d = CWeaponCraftParameters::JewelryCraftingToolDurability;              break;
+                       case ITEM_TYPE::MeleeWeaponTool:d = CWeaponCraftParameters::MeleeWeaponCraftingToolDurability;  break;
+                       case ITEM_TYPE::RangeWeaponTool:d = CWeaponCraftParameters::RangeWeaponCraftingToolDurability;  break;
+                       case ITEM_TYPE::ToolMaker:              d = CWeaponCraftParameters::ToolCraftingToolDurability;                 break;
 
                        default:
                                return 0;
```

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Looks ok.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Crafted tool durability, fix #156
